packages/page-accounts/src/modals/CreateEthDerivationPath.tsx

```diff
@@ -0,0 +1,159 @@
+// Copyright 2017-2021 @polkadot/app-accounts authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import BN from 'bn.js';
+import React, { ReactNode, useCallback, useEffect, useRef, useState } from 'react';
+
+import { Checkbox, Dropdown, Input, InputNumber, MarkError, MarkWarning, Modal } from '@polkadot/react-components';
+
+import { useTranslation } from '../translate';
+import { DeriveValidationOutput, PairType } from '../types';
+
+interface Props {
+  className?: string;
+  onChange: (string: string) => void;
+  seedType: string;
+  derivePath: string;
+  deriveValidation: DeriveValidationOutput | undefined;
+  seed: string;
+  addressFromSeed: (seed: string, derivePath: string, pairType: PairType) => string;
+}
+
+export const ETH_DEFAULT_PATH = "m/44'/60'/0'/0/0";
+
+function CreateEthDerivationPath ({ addressFromSeed,
+  className,
+  derivePath,
+  deriveValidation,
+  onChange,
+  seed,
+  seedType }: Props): React.ReactElement<Props> {
+  const { t } = useTranslation();
+  const [addIndex, setAddIndex] = useState(0);
+  const [customIndex, setCustomIndex] = useState(new BN(0));
+  const [addressList, setAddressList] = useState<{ key: number; text: ReactNode; value: number; }[]>([{
+    key: 0,
+    text: t('Address index 0'),
+    value: 0
+  }]);
+  const [useCustomPath, setUseCustomPath] = useState(false);
+  const [useCustomIndex, setUseCustomIndex] = useState(false);
+
+  const errorIndex = useRef<Record<string, string>>({
+    INVALID_DERIVATION_PATH: t<string>('This is an invalid derivation path.'),
+    PASSWORD_IGNORED: t<string>('Password are ignored for hex seed'),
+    SOFT_NOT_ALLOWED: t<string>('Soft derivation paths are not allowed on ed25519'),
+    WARNING_SLASH_PASSWORD: t<string>(
+      'Your password contains at least one "/" character. Disregard this warning if it is intended.'
+    )
+  });
+
+  const _toggleCustomPath = useCallback(() => {
+    setUseCustomPath(!useCustomPath);
+  }, [setUseCustomPath, useCustomPath]);
+
+  const _toggleCustomIndex = useCallback(() => {
+    setUseCustomIndex(!useCustomIndex);
+  }, [useCustomIndex, setUseCustomIndex]);
+
+  const _onChangeCustomIndex = useCallback((e: BN) => {
+    setCustomIndex(e);
+  }, [setCustomIndex]);
```

Comment:
I suppose this could be called directly, e.g. where used just pass through `setCustomIndex` since this does not apply any additional logic?